content/blog/the-rise-of-special-project-infra.md

```diff
@@ -0,0 +1,168 @@
+   1. Access the host over SSH.
+   1. Clone the configuration.
+   1. Run `nixos-rebuild switch` to build and activate the configuration (and make it the boot default).
+
+With this, we had a fully operational production server. For the finishing touches, we would make just a few program-specific tweaks.
+
+#### Streaming service configuration
+
+As previously mentioned, much of the Owncast configuration is stored as mutable state, enabling the service to be modified on-the-fly from the administrative web UI. A suggested procedure follows:
+
+1.  Log in to the administrative web UI:
+
+    - URL: `https://<target FQDN>/admin/` (e.g., `https://live.nixos.org/admin/`)
+    - User name: `admin`
+    - Password: `<stream key>` (`abc123` by default)
+
+1.  In server settings, set a new Stream Key. (Note that this is also the administrative web UI password.)
+
+1.  In general settings, modify instance details, tags, and social handles as appropriate.
+
+1.  In video settings:
+
+    - Add stream variants to enable adaptive bitrate streaming to accommodate users on various-quality networks. The following table may be used as a guideline:
+
+      | Encoder | Name | Resolution (WxH) |        Bitrate | Framerate    |
+      | ------- | ---- | ---------------- | -------------: | ------------ |
+      | x264    | SD   | 854x480          |  800-1200 kbps | 25 or 30 fps |
+      | x264    | HD   | 1280x720         | 1200-1900 kbps | 25 or 30 fps |
+      | x264    | FHD  | 1920x1080        | 1900-4500 kbps | 25 or 30 fps |
+
+      Note: some services recommend a higher bitrate for HD:
+
+      | Encoder | Name | Resolution (WxH) |   Bitrate | Framerate    |
+      | ------- | ---- | ---------------- | --------: | ------------ |
+      | x264    | HD   | 1280x720         | 3000 kbps | 25 or 30 fps |
+      | x264    | FHD  | 1920x1080        | 4500 kbps | 25 or 30 fps |
+
+      Note: our stress tests showed that the use of standard variants with our VPS were reportedly adequate for most users but problematic for some in Japan, Thailand, and the UK. Based on these results, object storage and CDN caching are recommended.
+
+      Additional notes:
+
+      - For interactive live streams (e.g., lectures with Q&A sessions), consider decreasing the latency buffer to Low.
+      - For further tuning, consider referring to a bitrate calculator (e.g., [Bitrate Calc](https://bitratecalc.com/)).
+
+#### Streaming client configuration
+
+With the server side all set, it was time to get streaming! The basic procedure follows:
+
+1. On a local workstation, install and launch [OBS Studio](https://obsproject.com/) ([obs-studio](https://search.nixos.org/packages?channel=22.05&show=obs-studio&from=0&size=50&sort=relevance&type=options&query=obs-studio) in Nixpkgs).
```

Comment:
```suggestion
1. On a local workstation, install and launch [OBS Studio](https://obsproject.com/) ([obs-studio](https://search.nixos.org/packages?channel=22.05&show=obs-studio&from=0&size=50&sort=relevance&type=options&query=obs-studio) in `nixpkgs`).
```

Reply:
I’m curious about the reasoning behind this one. Throughout our official documentation (including the Nixpkgs manual itself), “Nixpkgs” is generally written as a proper noun unless specifying, e.g., an arbitrary name for a channel or flake input, which is not the case here.

Reply:
First of all, do not consider the manuals authoritative, they've just grown randomly over 19 years and are all over the place with regard to writing style and even terminology.

We haven't figured this out yet, and here is the single source of truth where it should happen eventually, hopefully: https://github.com/NixOS/nix.dev/issues/263

My consideration is this: Nixpkgs is "not a word" because you can't properly read it like other words. You have to know it means something else, namely "Nix packages". `nixpkgs` does not have to be a word, it's a code token that describes something build in a computer language, it's meaning is the encoding it refers to. You don't have to "read" it, but there should be a canonical pronunciation which derives from the original meaning: Nix packages.

Since "Nix packages" is awkward to use as a subject ("X is in Nix packages", "Nix packages is a Y", and I think the thing should be singular as it describes a closed concept) I proposed "Nix package collection". The repository is still awkwardly described as "Nix packages collection" and I would like to change that once we fix terminology.

Reply:
Thanks for the explanation. I’ll go ahead and change all occurrences of “Nixpkgs” to “`nixpkgs`” for this article.